internal/rbd/replicationcontrollerserver.go

```diff
@@ -300,23 +300,56 @@ func getOperationName(poolName string, optName operation) string {
 // createDummyImage creates a dummy image as a workaround for the rbd
 // scheduling problem.
 func createDummyImage(ctx context.Context, rbdVol *rbdVolume) error {
+	var err error
+	var imgName string
 	optName := getOperationName(rbdVol.Pool, dummyImageCreated)
 	if _, ok := operationLock.Load(optName); !ok {
 		// create a dummy image
-		imgName, err := getDummyImageName(rbdVol.conn)
+		imgName, err = getDummyImageName(rbdVol.conn)
 		if err != nil {
 			return err
 		}
 		dummyVol := *rbdVol
 		dummyVol.RbdImageName = imgName
+		f := []string{
+			librbd.FeatureNameLayering,
+			librbd.FeatureNameObjectMap,
+			librbd.FeatureNameExclusiveLock,
+			librbd.FeatureNameFastDiff,
+		}
+		features := librbd.FeatureSetFromNames(f)
+		dummyVol.imageFeatureSet = features
+		// create 1MiB dummy image. 1MiB=1048576 bytes
+		dummyVol.VolSize = 1048576
 		err = createImage(ctx, &dummyVol, dummyVol.conn.Creds)
-		if err != nil && !strings.Contains(err.Error(), "File exists") {
-			return err
+		if err != nil {
+			if strings.Contains(err.Error(), "File exists") {
+				err = repairDummyImage(ctx, &dummyVol)
+			}
 		}
+	}
+
+	if err == nil {
```

Comment:
Although harmless outside the Load for the same, should this move inside the if block above?